src/components/PostDetails.tsx

```diff
@@ -1,118 +1,66 @@
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
-  };
+export const PostDetails: React.FC = memo(function PostDetailsComponent() {
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.posts);
+  const { comments, loaded, hasError, visibleForm } = useAppSelector(
+    state => state.comments,
+  );
 
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
-
-  const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
-  };
+  const deletedComment = useCallback(
+    (commentId: number) => {
+      dispatch(commentsActions.deleteComment(commentId));
+      dispatch(commentsActions.initCommentDeleted(commentId));
+    },
+    [dispatch],
+  );
 
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    // eslint-disable-next-line max-len
-    setComments(currentComments =>
-      currentComments.filter(comment => comment.id !== commentId),
-    );
+  const createComment = useCallback(
+    async ({ name, email, body }: CommentData) => {
+      dispatch(
+        commentsActions.initCommentCreat({
+          name,
+          email,
+          body,
+          id: 0,
+          postId: selectedPost?.id || 0,
+        }),
+      );
+    },
+    [dispatch, selectedPost?.id],
+  );
 
-    await commentsApi.deleteComment(commentId);
+  const show = {
+    commentsError: !loaded && hasError,
+    comments: !loaded && !hasError && !!comments.length,
+    noComments: !loaded && !hasError && !comments.length,
+    buttonWriteComment: !loaded && !hasError && !visibleForm,
   };
 
   return (
     <div className="content" data-cy="PostDetails">
       <div className="block">
-        <h2 data-cy="PostTitle">{`#${post.id}: ${post.title}`}</h2>
+        <h2 data-cy="PostTitle">{`#${selectedPost?.id}: ${selectedPost?.title}`}</h2>
 
-        <p data-cy="PostBody">{post.body}</p>
+        <p data-cy="PostBody">{selectedPost?.body}</p>
       </div>
 
       <div className="block">
-        {!loaded && <Loader />}
+        {loaded && <Loader />}
```

Comment:
it seems misleading. If you show loader when this variable is `true` then it should be named `loading` (or better `isLoading`). Also, consider using only one variable instead of both for almost the same purpose (loading, submitting)

Reply:
I think they will conflict during simultaneous asynchronous actions. (Also, consider using only one variable instead of both for almost the same purpose (loading, submitting))